crates/server/src/authority/catalog.rs

```diff
@@ -701,10 +701,11 @@ async fn build_authoritative_response(
             response_header.set_authoritative(true);
             Some(records)
         }
-        // This request was refused
         // TODO: there are probably other error cases that should just drop through (FormErr, ServFail)
-        Err(LookupError::ResponseCode(ResponseCode::Refused)) => {
-            response_header.set_response_code(ResponseCode::Refused);
+        Err(LookupError::ResponseCode(
+            rcode @ ResponseCode::Refused | rcode @ ResponseCode::NotAuth,
+        )) => {
```

Comment:
In the `Err(_)` arm below there are functions on `LookupError` used to determine the logical presence of specific response codes that consider other variants than `LookupError::ResponseCode` (like treating a `LookupError::ProtoError(p)` where `p.is_nx_domain()` as equiv to `LookupError::ResponseCode(ResponseCode::NXDomain)`.

Is the assumption here that the only codepaths that produce a Refused/NotAuth we're interested in are yielding `LookupError::ResponseCode` instances and so similar logic isn't required?